tokio-postgres/src/query.rs

```diff
@@ -50,13 +54,125 @@ where
     };
     let responses = start(client, buf).await?;
     Ok(RowStream {
-        statement,
+        statement: statement,
         responses,
         rows_affected: None,
         _p: PhantomPinned,
     })
 }
 
+enum QueryProcessingState {
```

Comment:
It feels like this is a lot of boilerplate over a few `match responses.next().await? { ... }` in a sequence.

Reply:
Fixed.